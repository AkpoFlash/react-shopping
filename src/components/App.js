import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch, withRouter } from 'react-router-dom';
import styled from 'styled-components';

import Menu from './Menu/Menu';
import Footer from './Footer/Footer.js';
import { setBooks } from '../actions/books';
import ShopWindow from './ShopWindow/ShopWindow';
import BookPage from './BookPage/BookPage';

const Container = styled.div`
	display: grid;
	grid-template-rows: 50px 1fr 50px;
	grid-template-columns: 25px 1fr 25px;
  grid-row-gap: 20px;
`;

const Content = styled.main`
  width: 100%;
  height: 100%;
  grid-column: 2;
  grid-row: 2;
  justify-items: center;
`;

class App extends Component {
  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  componentDidMount() {
    fetch('./books.json')
      .then( response => response.json())
      .then(data => this.props.setBooks(data));
  }

  render() {
    return (
      <Container>
        <Menu />
        <Content>

          <Switch>
            <Route path='/books' component={ ShopWindow } exact />
            <Route path='/books/:id' component={ BookPage } />
          </Switch>
          
        </Content>
        <Footer />
      </Container>
    );
  }
}

const mapStateToProps = ({ languages }) => ({
  usersLang: languages.usersLang,
});

const mapDispatchToProps = (dispatch) => ({
  setBooks: items => dispatch(setBooks(items)),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
