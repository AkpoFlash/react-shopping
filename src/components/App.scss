*{
  outline: 1px solid red;
}

:root{
  --secendaryColor: #{$colorGray};
}

body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.container {
	display: grid;
	grid-template-rows: 50px 25px 1fr 50px;
	grid-template-columns: 25px 1fr 25px;
	grid-row-gap: 20px;

	&__context {
    width: 100%;
    height: 100%;
    display: grid;
    grid-column: 2;
    grid-row: 3;
    justify-items: center;
    grid-template-columns: repeat(4, 1fr);
    grid-gap: 25px;
    grid-auto-columns: minmax(100px, 20%);
	}
}