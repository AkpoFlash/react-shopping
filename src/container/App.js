import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from 'semantic-ui-react';
import { Card } from 'semantic-ui-react';
import orderBy from 'lodash/orderBy';
import filter from 'lodash/filter';

import Menu from '../container/Menu';
import BookCard from '../container/BookCard';
import Filter from './Filter.js';
import { setBooks } from '../actions/books';

class App extends Component {
  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  componentDidMount(){
    fetch('/books.json')
      .then( response => response.json())
      .then( data => this.props.setBooks(data));
  }

  render() {
    const { books, isReady } = this.props;
    return (
      <Container>
        <Menu />
        <Filter />
        <div className="App">
          <Card.Group itemsPerRow={4}>
            { !isReady ? 'Загрузка...' :
              books.map( ( book, i ) => (
                <BookCard key={i} { ...book } />
              ))
            }
          </Card.Group>
        </div>
      </Container>
    );
  }
}

const sortBy = (books, filterBy, searchQuery) => {
  books = filter(books, (item) => isInclude(item, searchQuery));
  switch (filterBy) {
    case 'all':
      return books;
    case 'price_high':
      return orderBy(books, 'price', 'desc');
    case 'price_low':
      return orderBy(books, 'price', 'asc');
    case 'author':
      return orderBy(books, 'author', 'asc');
    default:
      return books;
  }
}

const mapStateToProps = ({ books, filter }) => ({
  books: sortBy(books.items, filter.filterBy, filter.searchQuery),
  isReady: books.isReady,
})

const mapDispatchToProps = (dispatch) => ({
  setBooks: items => dispatch(setBooks(items)),
})

const isInclude = (item, searchQuery) => {
  return item.title.toLowerCase().indexOf(searchQuery.toLowerCase()) >= 0 ||
    item.author.toLowerCase().indexOf(searchQuery.toLowerCase()) >= 0;
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
