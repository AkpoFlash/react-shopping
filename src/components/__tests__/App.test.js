import React from 'react';
import { render } from 'enzyme';

import App from '../App';

describe('App', () => {
	const app = render(App);

	it('should App not changed', () => {
		expect(app).toMatchSnapshot();
	});
});