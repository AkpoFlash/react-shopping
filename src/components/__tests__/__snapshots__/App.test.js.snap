// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should App not changed 1`] = `null`;
